internal/sdkio/ringbuffer_test.go

```diff
@@ -0,0 +1,369 @@
+package sdkio
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"testing"
+)
+
+func TestNew(t *testing.T) {
+	byteSlice := make([]byte, 10)
+	cases := map[string]struct {
+		expectedStart int
```

Comment:
Is this extra test case needed? It seems like the behavior of `New` would be captured by the `Write` and `Read` cases.